cmd/kubelet/app/server.go

```diff
@@ -297,8 +297,8 @@ func initConfigz(kc *componentconfig.KubeletConfiguration) (*configz.Config, err
 }
 
 func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
-	// TODO: this should be replaced by a --standalone flag
-	standaloneMode := (len(s.APIServerList) == 0 && !s.RequireKubeConfig)
+	// TODO: once server list is removed this line should be changed to  s.Standalone && !s.RequireKubeConfig
+	standaloneMode := s.Standalone && len(s.APIServerList) == 0 && !s.RequireKubeConfig
```

Comment:
This should be `(len(s.APIServerList) == 0 || s.Standalone) && !s.RequireKubeConfig` as `s.Standalone` will be false by default, and you still need to maintain legacy behavior where `len(s.APIServerList) == 0 && !s.RequireKubeConfig` is sufficient to enable standalone mode.

It might also be useful to return an error message like, "The Kubelet cannot run in standalone mode if RequireKubeConfig is set to true." for if someone explicitly sets both `RequireKubeConfig` and `Standalone` (but just let the Kubelet continue as usual with standalone disabled in the `len(s.APIServerList) == 0 && s.RequireKubeConfig == true` case, again we need to maintain legacy behavior.